### PR TITLE
Support vxWorks in libc-test

### PR DESCRIPTION
**Pull request by BaoshanPang**
r? @gnzlbg

### CODE REVIEW

**Comment by BaoshanPang** [comment]
cc @n-salim 
cc @maya-cm
cc @UmeshKalappa

File: libc-test/build.rs [hunk 1]
```diff
@@ -1927,6 +1928,157 @@ fn test_emscripten(target: &str) {
...
+    }
+
+    cfg.skip_struct(move |ty| {
+        match ty {
+            "passwd"      |
+            //the below definitions is part of public/private/ 
+            "_Vx_EVENTS_RSRC" |
+            "_Vx_semaphore" |
+            //Avaliable in Kernel space.
+            "epoll_event" => true,
```
Comment (gnzlbg) [inline libc-test/build.rs]: I thought `epoll_event` was removed from the libc crate ?

File: libc-test/build.rs [hunk 2]
```diff
@@ -1927,6 +1928,157 @@ fn test_emscripten(target: &str) {
...
+               "wchar.h",
+               "errno.h",
+    }
+
+    cfg.skip_struct(move |ty| {
+        match ty {
+            "passwd"      |
+            //the below definitions is part of public/private/ 
+            "_Vx_EVENTS_RSRC" |
+            "_Vx_semaphore" |
```
Comment (gnzlbg) [inline libc-test/build.rs]: Are these private in the libc crate ? If so and ctest picks them up then maybe ctest has a bug :/

File: libc-test/build.rs [hunk 3]
```diff
@@ -1927,6 +1928,157 @@ fn test_emscripten(target: &str) {
...
+        | "S_nfsLib_NFSERR_MLINK"
+        | "S_IFDEVMEM"
+        | "CHILD"
+        | "TIME_RELTIME"
+        | "RTLD_DEFAULT"
+        | "EIOA"
+        | "FIONBIO"
+        | "FIOWRITE"
+        | "FIONREAD"
+        | "VXSIM_EWOULDBLOCK" => true,
```
Comment (gnzlbg) [inline libc-test/build.rs]: Why can't these be fixed?

File: libc-test/build.rs [hunk 4]
```diff
@@ -1927,6 +1928,157 @@ fn test_emscripten(target: &str) {
...
+        | "lchown" | "setpgid" | "fstatat" | "setsid" | "tcgetpgrp"
+        | "tcsetpgrp" | "killpg" | "mlock" | "munlock" | "mlockall"
+        | "munlockall" | "mmap" | "munmap" | "if_nametoindex"
+        | "if_indextoname" | "flock" | "res_init" | "mknod" | "chroot"
+        | "getsid" | "tcdrain" | "tcflow" | "tcflush" | "tcgetsid"
+        | "tcsendbreak" | "mkdtemp" | "nice" | "grantpt" | "posix_openpt"
+        | "ptsname" | "unlockpt" | "strcasestr" | "socketpair"
+        | "pwrite64" | "pread64" | "fork" | "_pathIsAbsolute" | "execvp"
+        | "setgroups" | "signal" | "getpagesize" | "dlclose" | "dlsym"
+        | "dlerror" | "dlopen" | "setrlimit" | "getrlimit" => true,
```
Comment (gnzlbg) [inline libc-test/build.rs]: These are too many issues. What's wrong with, e.g., `mmap` ?

**Comment by BaoshanPang** [comment]
@gnzlbg 
Thanks for pointing out the problems, please help to review the latest code.

File: libc-test/build.rs [hunk 5]
```diff
@@ -1927,6 +1928,114 @@ fn test_emscripten(target: &str) {
...
+               "wchar.h",
+               "errno.h",
+               "sys/mman.h",
+               "pathLib.h",
+    }
+    cfg.skip_const(move |name| match name {
+        // sighandler_t weirdness
+        "SIG_DFL" | "SIG_ERR" | "SIG_IGN"
+        // This is not defined in vxWorks
+        | "RTLD_DEFAULT"   => true,
```
Comment (gnzlbg) [inline libc-test/build.rs]: Please add a FIXME here as well for now.

**Comment by BaoshanPang** [comment]
@gnzlbg 

Is it good enough for 'roll'?

**Comment by gnzlbg** [comment]
@bors: r+

-- 

Sorry it took so long, CI has been broken since friday till today.

**Pull request merged**